Algorithmia/CLI.py

```diff
@@ -0,0 +1,331 @@
+                else:
+                    file = client.file(f)
+
+                    if(file.exists()):
+                        result += file.getString()
+                    else:
+                        result = "file does not exist "+f
+                        break
+            else:
+                print("operands must be a path to a remote data source data://")
+                break
+
+        return result
+
+    # algo cp <src> <dest>
+    def cp(self, src, dest, client):
+
+        if(src is None or dest is None):
+            print("expected algo cp <src> <dest>")
+        else:
+
+            destLocation = client.file(dest)
+            for f in src:
+
+
+                #if dest is a directory apend the src name
+                #if there are multiple src files only the final one will be copied if dest is not a directory
+                destPath = dest
+
+                path = dest.split('/')
+                print(path)
+
+                if(os.path.isdir(dest) or client.dir(dest).exists() and len(path) <= 5):
+                    if(dest[-1] == '/' and path[-1] == ''):
+                        destPath+=client.file(f).getName()
+                    elif(len(path) == 4 or "data://" not in dest):
+                        destPath+='/'+client.file(f).getName()
+
+                if(f[-1] == '*'):
+                    src += ['data://'+file.path for file in client.dir(f[:len(f)-2]).files()]
+
+                #if src is local and dest is remote
+                elif("data://" not in f and "data://" in dest):
+                    print(destPath)
+                    client.file(destPath).putFile(f)
+
+                #if src and dest are remote
+                elif("data://" in f and "data://" in dest):
+                    file = client.file(f).getFile()
+                    filename = file.name
+                    file.close()
+
+                    client.file(destPath).putFile(filename)
+
+                #if src is remote and dest is local
+                elif("data://" in f and "data://" not in dest):
+                    file = client.file(f).getFile()
+                    filename = file.name
+                    file.close()
+                    print(destPath)
+                    os.system("mv " + filename + " " + destPath)
+                else:
+                    print("at least one of the operands must be a path to a remote data source data://")
+
+    def getconfigfile(self):
+        if(os.name == "posix"):
+            #if!windows
+            #~/.algorithmia/config
+            #create the api key file if it does not exist
+            keyPath = os.environ['HOME']+"/.algorithmia/"
+            keyFile = "config"
+            if(not os.path.exists(keyPath)):
+                os.mkdir(keyPath)
+                file = open(keyPath+keyFile,"w")
+                file.write("[profiles]")
+                file.close()
+
+            key = keyPath+keyFile
```

Comment:
As a general software engineering practice, I'd recommend doing as little copy-paste as we can, because it increases our possible maintenance load in the future. For instance, these lines are repeated in the `elif` block below, so they can be encapsulated in a function and be re-used.